db/dcrpg/pgblockchain.go

```diff
@@ -52,14 +56,63 @@ func makeAddressCounter() *addressCounter {
 	}
 }
 
+type TicketTxnIDGetter struct {
```

Comment:
@chappjc comment this to indicate it's a thread-safe cache for transaction id to DB row IDs, required by ChainDB for efficient random access to a transactions DB row IDs without having to query the table (by hash or whatever).  A cache miss will automatically call `RetrieveTicketIDByHash`